tests/foreman/api/test_notifications.py

```diff
@@ -0,0 +1,230 @@
+        login=gen_string("alphanumeric"),
+        password=gen_string("alphanumeric"),
+        mail='root@localhost',
+    ).create()
+    user.mail_enabled = True
+    user.update()
+
+    yield user
+
+    user.delete()
+
+
+@pytest.fixture
+def reschedule_long_running_tasks_notification(target_sat):
+    """Reschedule long-running tasks checker from midnight (default) to every minute.
+    Reset it back after the test.
+    """
+    default_cron_schedule = '0 0 * * *'
+    every_minute_cron_schedule = '* * * * *'
+
+    assert (
+        target_sat.execute(
+            f"FOREMAN_TASKS_CHECK_LONG_RUNNING_TASKS_CRONLINE='{every_minute_cron_schedule}' "
+            "foreman-rake foreman_tasks:reschedule_long_running_tasks_checker"
+        ).status
+        == 0
+    )
+
+    yield
+
+    assert (
+        target_sat.execute(
+            f"FOREMAN_TASKS_CHECK_LONG_RUNNING_TASKS_CRONLINE='{default_cron_schedule}' "
+            "foreman-rake foreman_tasks:reschedule_long_running_tasks_checker"
+        ).status
+        == 0
+    )
+
+
+@pytest.fixture
+def start_postfix_service(target_sat):
+    """Start postfix service (disabled by default). Stop it afterwards."""
+    assert target_sat.execute('systemctl start postfix').status == 0
+    yield
+    assert target_sat.execute('systemctl stop postfix').status == 0
+
+
+@pytest.fixture
+def clean_root_mailbox(target_sat):
+    """Backup & purge local mailbox of the Satellite's root@localhost user.
+    Restore it afterwards.
+    """
+    root_mailbox = '/var/spool/mail/root'
+    root_mailbox_backup = f'{root_mailbox}.bak'
+    target_sat.execute(f'cp -f {root_mailbox} {root_mailbox_backup}')
+    target_sat.execute(f'truncate -s 0 {root_mailbox}')
+
+    yield root_mailbox
+
+    target_sat.execute(f'mv -f {root_mailbox_backup} {root_mailbox}')
+
+
+@pytest.fixture
+def wait_for_incoming_mail(target_sat, clean_root_mailbox):
+    """Wait for mbox file to change in size, i.e., that a new e-mail arrived (in most cases)."""
+    timeout = 300
+    result = target_sat.execute(
+        f'TERM=xterm timeout {timeout} watch --no-title --chgexit stat -c %Z {clean_root_mailbox}'
+    )
+    assert result.status == 0, f'No new e-mail has arrived to {clean_root_mailbox} on Satellite.'
+    return True
+
+
+@pytest.fixture
+def root_mailbox_copy(target_sat, clean_root_mailbox, wait_for_incoming_mail):
+    """Parsed local system copy of the Satellite's root user mailbox.
+
+    :returns: :class:`mailbox.mbox` instance
+    """
+    assert wait_for_incoming_mail
+    result = target_sat.execute(f'cat {clean_root_mailbox}')
+    assert result.status == 0, f'Could not read mailbox {clean_root_mailbox} on Satellite host.'
+    mbox_content = result.stdout
+    _, local_mbox_file = mkstemp()
+    with open(local_mbox_file, 'w') as fh:
+        fh.writelines(mbox_content)
+    return mbox(path=local_mbox_file)
+
+
+@pytest.fixture
+def latest_email_message(target_sat, root_mailbox_copy):
```

Comment:
Here I'm fishing a bit, but are you certain that the message you are looking for will always be the last one? I'm thinking about some stale task that could get in the way, so maybe we could try to filter messages by task id

Reply:
Also, I think some other notification may come before the long-running-task notification. In that case, it wouldn't even be *any* of the notifications in the mailbox. I think it's necessary to wait for the specific notification, not just the first to come.

Reply:
Here it was a compromise between reliability and re-usability of the mailbox fixtures for possible other notification tests.
But in case of parallel test run, some other e-mail can indeed arrive there. 

I reworked the e-mail checks and now the test waits for the task ID to be found in the mailbox before proceeding further.